temporalio/worker/_workflow_instance.py

```diff
@@ -1733,6 +1749,7 @@ def _register_task(
 
     def _run_once(self, *, check_conditions: bool) -> None:
         try:
+            self._current_thread_id = threading.get_ident()
```

Comment:
I think this and the finally should be as part of `activate` not as part of `_run_once`. Thread and loop are unrelated IMO.